src/test/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTaskTest.java

```diff
@@ -377,6 +431,19 @@ public void testShouldNotThrowIfReporterDoesNotExist() {
     setUpTask();
   }
 
+  @Test
+  public void testShouldVerifyChangingTopic() {
+    assumeFalse(flushSynchronously);
+    setUpTask();
+    String changedTopic = "routed-to-another-topic";
+    SinkRecord record = record(changedTopic, false, false, 0);
+    when(context.assignment()).thenReturn(Collections.singleton(new TopicPartition("original-topic-name", 1)));
+    assertThrows(String.format("Found a topic name '%s' that doesn't match assigned partitions."
```

Comment:
@avocader  What we want to do is both things:

1. Check that exception is of type ConnectException
2. Also compare that message is equal so we know what type of ConnectException was thrown.

This is a better alternative.
```
ConnectException ce = assertThrows( () -> task.put(Collections.singletonList(record)));
assertEquals(String.format("Found a topic name '%s' that do...........", ce.getMessage());
```


if you see the implementation of assertThrows then its not comparing message, the message will be used to create an Error when types aren't equal.
```
public static <T extends Throwable> T assertThrows(String message, Class<T> expectedThrowable, ThrowingRunnable runnable) {
        try {
            runnable.run();
        } catch (Throwable var9) {
            if (expectedThrowable.isInstance(var9)) {
                return var9;
            }

            String expected = formatClass(expectedThrowable);
            Class<? extends Throwable> actualThrowable = var9.getClass();
            String actual = formatClass(actualThrowable);
            if (expected.equals(actual)) {
                expected = expected + "@" + Integer.toHexString(System.identityHashCode(expectedThrowable));
                actual = actual + "@" + Integer.toHexString(System.identityHashCode(actualThrowable));
            }

            String mismatchMessage = buildPrefix(message) + format("unexpected exception type thrown;", expected, actual);
            AssertionError assertionError = new AssertionError(mismatchMessage);
            assertionError.initCause(var9);
            throw assertionError;
        }

        String notThrownMessage = buildPrefix(message) + String.format("expected %s to be thrown, but nothing was thrown", formatClass(expectedThrowable));
        throw new AssertionError(notThrownMessage);
    }

```

Reply:
Thanks, good catch, I didn't realize `assertThrows()` doesn't check the exception message.